src/geometry/Sphere.ts

```diff
@@ -13,15 +14,10 @@ function sphereSignedDistFunc(coord: vec3): number {
     return Math.sqrt(coord[0] * coord[0] + coord[1] * coord[1] + coord[2] * coord[2]) - RADIUS;
 }
 
-// TODO(abhimadan): Create WorkingGeometry here instead
-export function genSphere(): BakedGeometry {
+export function genSphere(fillColor: Color = RGBColor.fromHex('#FF0000')): WorkingGeometry {
     const sphere = new ScalarField(DIM, LENGTH, sphereSignedDistFunc);
     const vertices = genIsoSurface(sphere);
+    const faces = range(vertices.length / 3).map((i: number) => new Face(range(i, i + 3)));
```

Comment:
your range here should be `range(i*3, (i+1)*3)`. Otherwise, we get `[0, 1, 2], [1, 2, 3], [2, 3, 4], ...`